reactor-core/src/main/java/reactor/core/scheduler/Schedulers.java

```diff
@@ -1553,4 +1557,51 @@ static final Object scanExecutor(Executor executor, Scannable.Attr key) {
 		return null;
 	}
 
+	static final void shutdownAndAwait(ExecutorService executor, Duration duration,
+			Sinks.Empty<Void> disposedNotifier) {
+		executor.shutdown();
+		Schedulers.parallel().schedule(() -> {
+			try {
+				boolean terminated = executor.awaitTermination(duration.toMillis(),
+						TimeUnit.MILLISECONDS);
+				if (terminated) {
+					disposedNotifier.tryEmitEmpty();
+				} else {
+					disposedNotifier.tryEmitError(new TimeoutException("Executor " +
+							"shutdown timed out"));
+				}
+			} catch (InterruptedException e) {
+				disposedNotifier.tryEmitError(e);
+			}
+		});
+	}
+
+	static final void shutdownAndAwait(ScheduledExecutorService[] executors,
```

Comment:
let me think a bit more about a way to get the best of both worlds: awaitTermination + isTerminated mix that consider a bunch of executors and awaits as little as possible, ideally only blocking a single thread. right now, we need to recreate as many threads as there are in the thread pool that is being shut down gracefully, just for the purpose of `awaitTermination`s 😢